docs/rules/saga-direct-delegation.md

```diff
@@ -0,0 +1,58 @@
+# Enforce using `yield*` to delegate to other sagas
+# saga-direct-delegation
+
+This rule enforces that delegation to sagas is done directly using `yield*`, rather than indirectly by plain `yield` of the returned iterator to the saga middleware.

```

Comment:
For `redux-tale`, I'm reasonably confident that the premise holds. It's quite straightforward to see that by looking at the code starting from its [generator driver loop](https://github.com/SaxoBank/redux-tale/blob/5cb1969fa46d0ea400008573a5e7a35293be8d86/src/index.js#L175) `runTask()` through to `resolveValue()`.

For `redux-saga`, I think it holds in the vast majority of cases. The exception, I think, being if you're using an [effect middleware](https://redux-saga.js.org/docs/api/#middleware-api) that does something with `yield`ed iterators (why ever do such a thing?). The output of saga monitor would also be different. The library is more complex, and there more code paths from its [generator driver loop](https://github.com/redux-saga/redux-saga/blob/f070f1bac9fdfdc2ce2e0a1c50f65101c793d695/packages/core/src/internal/proc.js#L64). Not being synchronous-first like tale doesn't seem to come into it.

- [ ] Add a "when not to use it" section, in the ESLint fashion

Reply:
867c1fd

Leaving conversation open so others can scrutinise.